### ISSUE TITLE
404 page not working properly

### ISSUE THREAD

**Issue by htdangkhoa** [rakkasjs/rakkasjs#149]
Hi there,

I'm using `rakkasjs` version `0.60.2`, but I think something is wrong when I try to implement a 404 page by the documentation.
In my video, 404 page it works 1 time when I change the page to `/upload`, but the 2nd time it should go to `/upload` instead of `/videos` (after full page reload).

I have tested all of the following situations where the situation occurs automatically:
- Using the Link component with `href`
- Using the Link component with the function `navigate`
- Using the anchor tag with the function `navigate`

https://github.com/rakkasjs/rakkasjs/assets/15341301/b02bf3ef-7430-409c-b2e3-64bd48561b48

**Comment by cyco130** [comment]
Hi!

Can you try version `0.7.0-next.17` (i.e. `rakkasjs@next`)? It includes a router fix that may have fixed the behavior you're encountering. I don't want to spend time backporting it since 0.7 release is just around the corner.

If the bug still continues, a small repro would be much appreciated.

**Comment by htdangkhoa** [comment]
Hi @cyco130, I appreciate your efforts, but I'm sorry to say that the problem still occurs in version `0.7.0-next.17`. But I just realized that it works fine with `StyledLink` component in both versions `0.60.2` and `0.7.0-next.17`

**Comment by cyco130** [comment]
Can you provide a small repro so I can understand what's going on?

**Comment by htdangkhoa** [comment]
@cyco130 This is my repo for this issue: https://github.com/htdangkhoa/rakkasjs-bug-link

**Comment by cyco130** [comment]
Great, I'm able to reproduce it now. It is indeed a bug, I'm gonna have a look today.

**Comment by cyco130** [comment]
I'll release version `0.7.0-next.18` later today that includes the fix.

But I noticed that you used `preventDefault` and called `navigate` yourself. That's not necessary since that's exactly what the `Link` component does anyway. If it's a relic from your previous attempts to make it work, it's OK. But I have a hunch that you're trying to load the 404 page _without a full reload_.

By design. Rakkas always does a full reload  before rendering a 404 page because the frontend router is not aware of all routes. There might be static files or API routes that would match but the frontend wouldn't know about them. So it always does a full reload to give them a chance to load.

If you don't agree with this behavior and want to show a 404 page without a full reload for some reason, you can use a catch-all route (e.g. `[...path].page.tsx`) and manually set the status to 404 like this:

```ts, name
import type { ResponseHeadersProps } from "rakkasjs";

const PageNotFound = () => {
  return <div>PageNotFound</div>;
};

export default PageNotFound;

export function headers(): ResponseHeadersProps {
  return { status: 404 };
}
```

This works because catch-all routes match before 404 pages. It also means that it's usually not a good idea to use root-level catch-all routes because they can shadow API routes and other routes.

**Comment by cyco130** [comment]
Released in `0.7.0-next.18`.

**Comment by htdangkhoa** [comment]
@cyco130 thanks for your work
